CMakeLists.txt

```diff
@@ -0,0 +1,4 @@
+cmake_minimum_required(VERSION 3.9)
```

Comment:
Any reason to require a cmake version this recent? Could be good to allow users on older cmake versions to use this as well.

Reply:
how about using 3.12? This is the version I use in the pipes library